### DISCUSSION TITLE
Automatically import scripts when linking .blend files

### DISCUSSION THREAD

**Discussion by HolandonHdunto** [UPBGE/upbge#1687] [Ideas/Features for UPBGE 0.2.5+]
An annoying issue with upbge and linking .blend files, upbge by default does not link scripts from files. A very bad thing mainly because it has to constantly go through the file path to import the script.
 Another thing is the unexpected closing of the program when running the game, mainly in a complex scene with instant loading that causes a crash mainly when using .blend files as PREFABS, that is, producing and programming separate entities that are later linked in a main file using groups. although the PREFABS is not very viable, mainly due to the difficulty of testing the interaction of an entity with the environment and with other entities.

**Comment by BluePrintRandom** [comment]
HolandonHdunto please open your own reports* stop spamming reports with mixed issues*

1 report = 1 issue*

**Comment by BluePrintRandom** [comment]
one way to link scripts is to import them in game and to preserve your file structure yourself in your shipped game*

we can load data blocks right out of blend files now / convert linked objects to KX_objects using scene.convert_blenderObject
(so you can on the fly create a bpy object - add a logic brick - add a text block - set it to call it - and convert it)

**Comment by BluePrintRandom** [comment]
Upbge is updated constantly, there is many new possibilities using geometry
nodes / openXR and more.

Please stop ranting here*

**Reply by HolandonHdunto** [reply]
The lack of update refers to the BGE of the 2.7x blender. Not from Upbge 0.3 or 0.2.x.

**Comment by BluePrintRandom** [comment]
If you want 2.7 updated you will have to learn openGL, C and py.

The original maintenance crew said its not worth saving (too much code rot
/ too much intellectual debt)

If you want to make better games in 2.7 (more achievable) learn glsl and
extend the engine in py / openGL.

Search for hg1 tutorials.
